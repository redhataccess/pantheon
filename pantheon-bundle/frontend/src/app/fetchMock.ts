if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}