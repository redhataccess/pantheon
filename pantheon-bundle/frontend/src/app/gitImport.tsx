import React, { useContext, useState } from "react";
import { Bullseye, Button, Alert, Modal, AlertActionCloseButton, FormGroup, TextInput, ModalVariant } from "@patternfly/react-core";
import "@app/app.css";
import { Redirect } from "react-router-dom"
import { GitImportContext } from "@app/contexts/GitImportContext"


export default function GitImport(props: any) {


  const [branch, setBranch] = useState("")
  const [isFormSubmitted, setIsFormSubmittedput] = useState(false)
  const [isMissingFields, setIsMissingFields] = useState(false)
  const [redirect, setStateRedirect] = useState(false)
  const [repository, setRepository] = useState("")

  const handleRepoInput = repository => {
    setRepository(repository);
  };

  const handleBranchInput = branch => {
    setBranch(branch);
  };

  const setRedirect = () => {
    setStateRedirect(true);
  };

  const renderRedirect = () => {
    if (redirect) {
      return <Redirect to="/" />
    } else {
      return ""
    }
  }

  const dismissNotification = () => {
    setIsMissingFields(false);
    setIsFormSubmittedput(false);
  };

  const context = useContext(GitImportContext);

  const { cloneRepoGit2Pantheon, status } = context

  const handleSubmit = () => {
    if (repository === "") {
      setIsMissingFields(true)
    } else {
      setIsFormSubmittedput(true)
      cloneRepoGit2Pantheon(branch, repository)
    }
  }

  return (
    <React.Fragment>
      <Bullseye>
        <div className="app-container">
          {isFormSubmitted && (
            <Modal
              variant={ModalVariant.small}
              title="Request submitted."
              isOpen={true}
              onClose={dismissNotification}
              actions={[<Button key="yes" variant="primary" onClick={setRedirect}>Yes</Button>,
              <Button key="no" variant="secondary" onClick={dismissNotification}>No</Button>]}>
              The git import has been submitted and the upload will begin. Do you want to be redirected to the module library?
            </Modal>)}
          <div>
            {isMissingFields && (
              <div className="notification-container">
                <Alert
                  variant="warning"
                  title="A repository url is required."
                  actionClose={<AlertActionCloseButton onClose={dismissNotification} />}
                />
              </div>
            )}
            <FormGroup
              label="Repository URL"
              fieldId="repository-url"
            >
              <TextInput id="repository-url" type="text" placeholder="Repository" value={repository} onChange={handleRepoInput} />
            </FormGroup>
            <br />
            <FormGroup
              label="Branch"
              fieldId="branch-name"
            >
              <TextInput id="branch-name" type="text" placeholder="Branch" value={branch} onChange={handleBranchInput} />
            </FormGroup>
            <br />
            <Button aria-label="Submit the repository and branch information to the git integration service." onClick={handleSubmit}>Submit</Button>
            <div>
              {renderRedirect()}
            </div>
          </div>
        </div>
      </Bullseye>
    </React.Fragment>
  );
}