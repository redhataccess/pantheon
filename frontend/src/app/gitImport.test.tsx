import React from 'react'
import { GitImport } from '@app/gitImport'
import '@app/fetchMock'

import { mount, shallow } from 'enzyme'
import { Bullseye, TextInput, FormGroup, Button } from '@patternfly/react-core'

describe('Login tests', () => {
  test('should render Login component', () => {
    const view = shallow(<GitImport />)
    expect(view).toMatchSnapshot()
  })

  it('should render a Bullseye layout', () => {
    const wrapper = mount(<GitImport />)
    const bullseyeLayout = wrapper.find(Bullseye)
    expect(bullseyeLayout.exists()).toBe(true)
  })

  it('should render a form group', () => {
    const wrapper = mount(<GitImport />)
    const formGroup = wrapper.find(FormGroup)
    expect(formGroup.exists()).toBe(true)
  })

  it('should render a text input', () => {
    const wrapper = mount(<GitImport />)
    const textInput = wrapper.find(TextInput)
    expect(textInput.exists()).toBe(true)
  })

  it('should render a Button', () => {
    const wrapper = mount(<GitImport />)
    const button = wrapper.find(Button)
    expect(button.exists()).toBe(true)
  })

})
