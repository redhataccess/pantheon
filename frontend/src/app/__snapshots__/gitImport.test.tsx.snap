// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests should render Login component 1`] = `
<Fragment>
  <Bullseye
    className=""
    component="div"
  >
    <div
      className="app-container"
    >
      <div>
        <FormGroup
          className=""
          fieldId="repository-url"
          isInline={false}
          isRequired={false}
          isValid={true}
          label="Repository URL"
        >
          <TextInput
            aria-label={null}
            className=""
            id="repository-url"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            placeholder="Repository"
            type="text"
            value=""
          />
        </FormGroup>
        <br />
        <FormGroup
          className=""
          fieldId="branch-name"
          isInline={false}
          isRequired={false}
          isValid={true}
          label="Branch"
        >
          <TextInput
            aria-label={null}
            className=""
            id="branch-name"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            placeholder="Branch"
            type="text"
            value=""
          />
        </FormGroup>
        <br />
        <Button
          aria-label="Submit the repository and branch information to the git integration service."
          className=""
          component="button"
          isActive={false}
          isBlock={false}
          isDisabled={false}
          isFocus={false}
          isHover={false}
          isInline={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Submit
        </Button>
        <div />
      </div>
    </div>
  </Bullseye>
</Fragment>
`;
